manifest.yaml

```diff
@@ -164,6 +164,18 @@ postprocess:
      # NB: we don't use -f here so we break when this is no longer needed
      rm -v /etc/iscsi/initiatorname.iscsi
 
+  # Add content_manifests dir to have json files for build info
+  # See https://issues.redhat.com/browse/GRPA-3731
+  - |
+     #!/usr/bin/env bash
+     REPOS=`jq .repos /usr/share/rpm-ostree/treefile.json`
+     OSTREE_VERSION=$(grep -oP '(?<=^OSTREE_VERSION=).+' /etc/os-release | tr -d "'")
+     ARCH=`jq .basearch /usr/share/rpm-ostree/treefile.json | tr -d '"'`
+     RHEL_VERSION=$(grep -oP '(?<=^RHEL_VERSION=).+' /etc/os-release | tr -d '"')
+     OS_NAME=$(grep -oP '(?<=^ID=).+' /etc/os-release | tr -d '"')
+     
+     mkdir -p /var/roothome/buildinfo/content_manifests && cd "$_" && touch $OS_NAME-$OSTREE_VERSION.json
```

Comment:
We can't write to `/var` at compose time. Maybe `/usr/share/buildinfo/content_manifests`? I recall we talked about adding a `tmpfiles.d` dropin if we really need it available under `/root`.

Reply:
My bad, I assumed we could write to` /var`and not to `/root`. I will go with `/usr/share/buildinfo/content_manifests` for now. Thanks:)

Reply:
Where can we ask these security scanners to change?  Using `/root` for this just conflicts with ostree today, and actually more generally would be problematic for many "image based updates" models.

Even having traditional `yum/rpm` write to `/root` is pretty ugly IMO.

Reply:
> Where can we ask these security scanners to change? Using `/root` for this just conflicts with ostree today, and actually more generally would be problematic for many "image based updates" models.
> 
> Even having traditional `yum/rpm` write to `/root` is pretty ugly IMO.

I made the assumption that it was a security scanner, but in the original ticket there is a reference to just `scanners`, which I guess could be anything doing any kind of scanning.

Since this request came from RHT ProdSec, we'd have to follow-up with them in https://issues.redhat.com/browse/GRPA-3731

Reply:
Ah, actually it is security scanners - `There is significant interest in getting this feature added to RHCOS from our security scanning partners.`

Reply:
I think it is fine to start working to change the security scanners, though I don't know how fast we can make that happen. Let's work with ProdSec to understand which scanners are looking for this file and see if we can get them modified.

Additionally, @ashcrow pointed out that the location of the file comes from OSBS [1][2], so we could work with OSBS to modify the location of the JSON file. Or suggest a secondary location if the first location isn't available.

But given the nature of this request, I think we have to work within the constraints of the existing tools, so we should pursue the use of `tmpfiles.d` to land the file in the location desired.

[1] https://github.com/containerbuildsystem/atomic-reactor/blob/master/atomic_reactor/constants.py#L200
[2] https://osbs.readthedocs.io/en/latest/users.html#image-content-manifests

Reply:
There's a reason though that ostree designs the constraints it does around `/usr` and `/var` etc.  We can certainly use `tmpfiles.d` or a custom systemd unit or whatever to write this content in `/var`.

But the thing is - that means that `rpm-ostree rollback` will mean the data in `/var` is out of sync with `/usr`.  

This doesn't matter a huge amount for containers because the larger container ecosystem doesn't support in-place offline updates - it's all about either running `yum update` (in place, online), or just tearing down the whole container and starting a new one (Kubernetes native style).

As I commented on the ticket, because rpm-ostree natively sticks this data in the commit object today, it is versioned alongside the deployment, which means that one can easily do e.g. diffs between them, and `rpm-ostree rollback` will always have the right data.

(Also, data in `/var` is mutable by default, whereas clearly this should be immutable)

Reply:
To clarify, the proposal here is to make e.g. `/root/buildinfo` be a symlink to `/usr/share/buildinfo`, not to copy.
Agree though that ideally we get scanners and other producers to use the data that's already available in commit metadata.

Reply:
Thanks for commenting on the Jira ticket with your input. I think we all completely missed the use of `rpmostree.rpmmd-repos` as a solution to this problem.

However, since the current tooling is looking for the information in a specific location, I still think we have to provide the data there...somehow.

Perhaps this is hacky, but might provide a solution given the constraints:

1. postprocess script that writes out the desired JSON file at `/usr/share/buildinfo/content_manifests/<nvr>.json`
2. a systemd unit that runs at boot which ensures there is a `/var/roothome/buildinfo/content_manifests/` directory
3. same systemd unit creates a symlink from  `/usr/share/buildinfo/content_manifests/<nvr>.json` to `/var/roothome/buildinfo/content_manifests/<nvr>.json`

WDYT?

Reply:
I think the location of the content_manifests/<nvr>.json does not have to be in /root/buildinfo, and /usr/share is fine. We can talk to our security scanning partners about the location once the content is ready.

Also, I don't think we can use the rpmostree.rpmmd-repos as is. We need to translate those repositories names to something in pulp, like the the repositories listed [here](https://access.redhat.com/security/data/metrics/repository-to-cpe.json), or directly to CPE (from Errata tool).